scripts/generate_hamilton_input_qpcr_dilution.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+import re
+import sys
+
+from EPPs.common import GenerateHamiltonInputEPP, InvalidStepError
+
+
+class GenerateHamiltonInputQPCRDilution(GenerateHamiltonInputEPP):
+    # define the column headers that will be used in the Hamilton input file and add to the csv_array to be
+    # used to write the file
+    csv_column_headers = ['Input Plate', 'Input Well', 'Output Plate', 'Output Well', 'Sample Volume', 'Buffer Barcode',
+                          'Buffer Volume']
+
+    # Define the output file
+    output_file_name = 'QPCR DILUTION.csv'
+
+    # Define the number of input containers that are permitted
+    _max_nb_input_containers = 1
+
+    # Define the number of output containers that are permitted
+    _max_nb_output_containers = 1
+
+    def calculate_rsb_volume(self, input_art):
+        """
+        The purpose of this function is to calculate the required volume of buffer to be added to a set volume of a DCT sample
+        to achieve a target concentration. The dilution will only occur if the DCT sample concentration is above a threshold
+         concentration. The script will also populate the output udfs of the step with a number input artifact udfs that
+         will get overwritten when the QPCR is re-run. This means the values are permanently stored and can be retried in
+          future.
+        """
+        output = self.process.outputs_per_input(input_art.id, ResultFile=True)[0]
+        if None not in self.udf_names:
+            for name in self.udf_names:
+                output.udf[name] = input_art.udf[name]
+        else:
+            raise InvalidStepError('UDF population failed due to missing input value in %s' % self.udf_names)
```

Comment:
udf_names is hard-coded and does not contain `None`

Reply:
Updated error handling for udf population.